### ISSUE TITLE
Calculation of hydrogen chain energy using the MDF method

### ISSUE THREAD

**Issue by JinTianZhongWuChiSheiNe** [pyscf/pyscf#2107]
Hello, I am using HF to calculate the energy of one-dimensional hydrogen chain. The energy by using AFTDT method is close to the literature, but the results by using DF and MDF methods are far from the actual value. I tried to use the result of the AFTDT method as the initial guess for the MDF method, but it still failed. Are there anything obviously wrong in my input? Thanks.
```python
import numpy
from pyscf import scf
from pyscf.pbc import df as pdf
from pyscf.pbc import scf as pbchf 
from pyscf.pbc import gto as pbcgto
from pyscf.pbc import tools
e = []
L = 1.8
cell = pbcgto.Cell()
cell.build(unit = 'A',
           a = [[L,0,0],[0,100,0],[0,0,100]],
           mesh = [40,20,20],
           atom = 'H 0 0 0; H 0.9 0 0',
           dimension=1, 
           verbose = 4,
           basis = 'gth-szv',
           pseudo = 'gth-pbe')
mf = pbchf.KRHF(cell)
mf.with_df = pdf.AFTDF(cell)
mf.kpts = cell.make_kpts([4,1,1])
e.append(mf.kernel())
dm_init_guess = mf.make_rdm1()

mf = pbchf.KRHF(cell).density_fit(auxbasis='weigend')
mf.kpts = cell.make_kpts([4,1,1])
e.append(mf.kernel())

mf = pbchf.KRHF(cell).mix_density_fit(auxbasis='weigend')
mf.init_guess = dm_init_guess
mf.kpts = cell.make_kpts([4,1,1])
e.append(mf.kernel())

# mol = tools.super_cell(cell, [4,1,1]).to_mol()
# mf = scf.RHF(mol)
# e.append(mf.kernel()/4)

print('1D: AFT      DF       MDF')
print(e)
```
Results:
AFT: -1.0123035727096745
DF: -11516.569579106246
MDF: -11516.566757778026]

**Comment by sunqm** [comment]
Please use `dimension=2` with one large vacuum in y-axis for now. In DF and MDF, there requires a cancellation between integrals computed with two different methods. Some improper parameter estimations probably cause one integral method singular. It is not easy to fix this problem in near future.

**Comment by JinTianZhongWuChiSheiNe** [comment]
Thank you very much! I will try it.

**Comment by sunqm** [comment]
This issue should have been resolved in pyscf-2.6
